### PR TITLE
Recipe for bertrand design by contract

### PR DESCRIPTION
**Pull request by bernedom**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by gocarlos — APPROVED** [review]

File: recipes/bertrand/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            self._cmake = CMake(self)
+            self._cmake.definitions["BERTRAND_BUILD_TESTING"] = False
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+        elif not self._supports_cpp17():
+            raise ConanInvalidConfiguration("bertrand requires C++17 support")
```
Comment (SpaceIm) [inline recipes/bertrand/all/conanfile.py:49]: Please don't use this pattern. Allow unknow compilers.

File: recipes/bertrand/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        self.info.header_only()
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "bertrand-{}".format(self.version)
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
```
Comment (prince-chrismc) [inline recipes/bertrand/all/conanfile.py:61]: build  and header only?

File: recipes/bertrand/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "bertrand"
+        self.cpp_info.names["cmake_find_package_multi"] = "bertrand"
```
Comment (prince-chrismc) [inline recipes/bertrand/all/conanfile.py:71]: That's the default value

```suggestion
```

File: recipes/bertrand/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(bertrand REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} bertrand_package_test.cpp)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
```
Comment (prince-chrismc) [inline recipes/bertrand/all/test_package/CMakeLists.txt:10]: this feature was added in 3.8 but im not sure 17 was a defined value until later

File: recipes/bertrand/all/test_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (prince-chrismc) [inline recipes/bertrand/all/test_package/conanfile.py:1]:
```suggestion
```

File: recipes/bertrand/all/test_package/CMakeLists.txt (hunk 4 above)
Comment (SpaceIm) [reply recipes/bertrand/all/test_package/CMakeLists.txt:10]: 3.8 is fine for C++17

File: recipes/bertrand/all/conanfile.py (hunk 1 above)
Comment (bernedom) [reply recipes/bertrand/all/conanfile.py:49]: So what is the correct way to inform the user that the library is C++17 only? 
As `cppstd` is still experimental relying on it being set correctly seems a bit risky to me.

I'm checking for `cppstd` above but on a soft way
```
        if self.settings.compiler.get_safe("cppstd"):
            tools.check_min_cppstd(self, "17")
````

File: recipes/bertrand/all/conanfile.py (hunk 2 above)
Comment (bernedom) [reply recipes/bertrand/all/conanfile.py:61]: Right will remove it

File: recipes/bertrand/all/conanfile.py (hunk 3 above)
Comment (bernedom) [reply recipes/bertrand/all/conanfile.py:71]: Will remove it

File: recipes/bertrand/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/bertrand/all/conanfile.py:49]: Check known compilers, allow unknown compilers and display a warnging. Here is an example:
https://github.com/conan-io/conan-center-index/blob/563bf4416f2d76392ab483e5300486759567b160/recipes/spy/all/conanfile.py#L28

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/bertrand/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+import os
+
+
+class SiConan(ConanFile):
+    name = "bertrand"
+    license = "LGPLv3"
```
Comment (uilianries) [inline recipes/bertrand/all/conanfile.py:10]:
```suggestion
    license = "LGPL-3.0-or-later"
```

Follow short-term name https://spdx.org/licenses/LGPL-3.0-or-later.html

**Review by uilianries — CHANGES_REQUESTED** [review]
Only a small detail and we are good to go.

File: recipes/bertrand/all/conanfile.py (hunk 6 above)
Comment (prince-chrismc) [reply recipes/bertrand/all/conanfile.py:10]: Hmmm https://opensource.org/licenses/LGPL-3.0 uses a different one

Comment (prince-chrismc) [reply recipes/bertrand/all/conanfile.py:10]: https://github.com/conan-io/conan-center-index/search?p=2&q=%22LGPL%22 Theres a bit too much variety

Comment (uilianries) [reply recipes/bertrand/all/conanfile.py:10]: LGPL-3.0 is deprecated, the new forms are LGPL-3.0-or-later and LGPL-3.0-only. There is a small section in the license which differs both. SDPX (short-term) is the correct pattern.

File: recipes/bertrand/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
```
Comment (uilianries) [inline recipes/bertrand/all/conanfile.py:3]:
```suggestion
```

Unused. You are calling tools.Version, not Version()

**Review by uilianries — CHANGES_REQUESTED** [review]

**Review by SpaceIm — COMMENTED** [review]
There is no build method, therefore caching CMake is useless (but doesn't hurt).

**Review by uilianries — DISMISSED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**